x/mongo/driver/topology/pool_test.go

```diff
@@ -773,6 +775,45 @@ func TestPool(t *testing.T) {
 			assert.Equalf(t, 2, p.totalConnectionCount(), "pool should have 2 total connection")
 			assert.Equalf(t, 0, p.availableConnectionCount(), "pool should have 0 idle connection")
 
+			p.close(context.Background())
+		})
+		t.Run("cancel wait queue", func(t *testing.T) {
+			t.Parallel()
+
+			cleanup := make(chan struct{})
+			defer close(cleanup)
+			addr := bootstrapConnections(t, 1, func(nc net.Conn) {
+				<-cleanup
+				_ = nc.Close()
+			})
+
+			p := newPool(poolConfig{
+				Address:     address.Address(addr.String()),
+				MaxPoolSize: 1,
+			})
+			err := p.ready()
+			noerr(t, err)
+
+			// check out first connection.
+			_, err = p.checkOut(context.Background())
+			noerr(t, err)
+
+			// Set a short timeout and check out again.
```

Comment:
```suggestion
			// Use a canceled context to check out another connection.
```
I don't think you need the goroutine to cancel the context, can we just do:
```
cancelCtx, cancel := context.WithCancel(context.Background())
cancel()
_, err = p.checkOut(cancelCtx)
```

Reply:
[ef458c5](https://github.com/mongodb/mongo-go-driver/pull/1058/commits/ef458c51dfccd3237a7d3454669196014aabeb1d)